metricbeat/module/elasticsearch/elasticsearch.go

```diff
@@ -304,6 +304,19 @@ func GetClusterState(http *helper.HTTP, resetURI string, metrics []string) (maps
 	return clusterState, err
 }
 
+func GetIndexSettings(http *helper.HTTP, resetURI string, indexPattern string) (mapstr.M, error) {
+	indicesSettingsURI := indexPattern + "/_settings"
```

Comment:
Would it make sense to add a `filter_path=` here to only request the fields necessary? It's a recurrent comment by the ES team when hitting APIs that can potentially return a lot of data

Reply:
It actually worsens performance 😅 
I had it a couple of commits ago but we were losing ~100ms compared to baseline.
So it would be a tradeoff between speed and data volume.

Reply:
I think we can sacrifice a few ms latency in order to not have to put a burden on the cluster to compute stuff we don't need and having to transport it all over the network to finally ditch most of it. What do you think?